core/src/repair/repair_service.rs

```diff
@@ -153,17 +218,68 @@ impl RepairTiming {
         dump_slots_elapsed: u64,
         get_votes_elapsed: u64,
         add_votes_elapsed: u64,
+        purge_outstanding_repairs: u64,
+        handle_popular_pruned_forks: u64,
         build_repairs_batch_elapsed: u64,
         batch_send_repairs_elapsed: u64,
     ) {
         self.set_root_elapsed += set_root_elapsed;
         self.dump_slots_elapsed += dump_slots_elapsed;
         self.get_votes_elapsed += get_votes_elapsed;
         self.add_votes_elapsed += add_votes_elapsed;
+        self.purge_outstanding_repairs += purge_outstanding_repairs;
+        self.handle_popular_pruned_forks += handle_popular_pruned_forks;
         self.build_repairs_batch_elapsed += build_repairs_batch_elapsed;
         self.batch_send_repairs_elapsed += batch_send_repairs_elapsed;
         self.send_repairs_elapsed += build_repairs_batch_elapsed + batch_send_repairs_elapsed;
     }
+
+    fn report(&self) {
+        datapoint_info!(
+            "repair_service-repair_timing",
+            ("set-root-elapsed", self.set_root_elapsed, i64),
+            ("dump-slots-elapsed", self.dump_slots_elapsed, i64),
+            ("get-votes-elapsed", self.get_votes_elapsed, i64),
+            ("add-votes-elapsed", self.add_votes_elapsed, i64),
+            (
+                "purge-outstanding-repairs",
```

Comment:
this is 1 of 2 new metrics added. Everything else should just be cosmetic change